server.go

```diff
@@ -524,15 +532,13 @@ func (c *serverConn) run(sctx context.Context) {
 			// TODO(stevvooe): Not wildly clear what we should do in this
 			// branch. Basically, it means that we are no longer receiving
 			// requests due to a terminal error.
-			recvErr = nil // connection is now "closing"
-			if err == io.EOF || err == io.ErrUnexpectedEOF {
+			if err == io.EOF || err == io.ErrUnexpectedEOF || errors.Is(err, syscall.ECONNRESET) {
 				// The client went away and we should stop processing
 				// requests, so that the client connection is closed
 				return
 			}
-			if err != nil {
-				logrus.WithError(err).Error("error receiving message")
-			}
+			logrus.WithError(err).Error("error receiving message")
```

Comment:
Why remove the nil check though? If this message is needed, then perhaps we need to change `sendStatus` to return error to send on `recvError`. 

It seems to me though that this is trying to catch the shutdown case which this block is purposefully avoiding when there is active connections. If the return after this line is necessary, then the shutdown logic in this goroutine still doesn't seem correct.

Reply:
As code at https://github.com/containerd/ttrpc/blob/v1.1.0/server.go#L363  will do the nil check and only send non-nill error to the `recvErr` channel, I think it's safe to remove the nil check here.

> It seems to me though that this is trying to catch the shutdown case which this block is purposefully avoiding when there is active connections. If the return after this line is necessary, then the shutdown logic in this goroutine still doesn't seem correct.

Yes, the return here is not necessary and should be done by the shutdown logic. I‘ll remove it, thx